config/confighttp/confighttp_test.go

```diff
@@ -156,6 +156,121 @@ func TestAllHTTPClientSettings(t *testing.T) {
+	host := &mockHost{
+		ext: map[component.ID]component.Component{
+			component.NewID("testauth"): &authtest.MockClient{ResultRoundTripper: &customRoundTripper{}},
+		},
+	}
+
+	maxIdleConns := 50
+	maxIdleConnsPerHost := 40
+	maxConnsPerHost := 45
+	idleConnTimeout := 30 * time.Second
+	tests := []struct {
+		name               string
+		settings           HTTPClientSettings
+		customRoundTripper func(next http.RoundTripper) (http.RoundTripper, error)
+		shouldError        bool
+	}{
+		{
+			name: "all_valid_settings",
+			settings: HTTPClientSettings{
+				Endpoint: "localhost:1234",
+				TLSSetting: configtls.TLSClientSetting{
+					Insecure: false,
+				},
+				ReadBufferSize:      1024,
+				WriteBufferSize:     512,
+				MaxIdleConns:        &maxIdleConns,
+				MaxIdleConnsPerHost: &maxIdleConnsPerHost,
+				MaxConnsPerHost:     &maxConnsPerHost,
+				IdleConnTimeout:     &idleConnTimeout,
+				Compression:         "",
+				DisableKeepAlives:   true,
+			},
+			customRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) { return next, nil },
+			shouldError:        false,
+		},
+		{
+			name: "all_valid_settings_with_none_compression",
+			settings: HTTPClientSettings{
+				Endpoint: "localhost:1234",
+				TLSSetting: configtls.TLSClientSetting{
+					Insecure: false,
+				},
+				ReadBufferSize:      1024,
+				WriteBufferSize:     512,
+				MaxIdleConns:        &maxIdleConns,
+				MaxIdleConnsPerHost: &maxIdleConnsPerHost,
+				MaxConnsPerHost:     &maxConnsPerHost,
+				IdleConnTimeout:     &idleConnTimeout,
+				Compression:         "none",
+				DisableKeepAlives:   true,
+			},
+			customRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) { return next, nil },
+			shouldError:        false,
+		},
+		{
+			name: "all_valid_settings_with_gzip_compression",
+			settings: HTTPClientSettings{
+				Endpoint: "localhost:1234",
+				TLSSetting: configtls.TLSClientSetting{
+					Insecure: false,
+				},
+				ReadBufferSize:      1024,
+				WriteBufferSize:     512,
+				MaxIdleConns:        &maxIdleConns,
+				MaxIdleConnsPerHost: &maxIdleConnsPerHost,
+				MaxConnsPerHost:     &maxConnsPerHost,
+				IdleConnTimeout:     &idleConnTimeout,
+				Compression:         "gzip",
+				DisableKeepAlives:   true,
+			},
+			customRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) { return next, nil },
+			shouldError:        false,
+		},
+		{
+			name: "error_round_tripper_returned",
+			settings: HTTPClientSettings{
+				Endpoint: "localhost:1234",
+				TLSSetting: configtls.TLSClientSetting{
+					Insecure: false,
+				},
+				ReadBufferSize:  1024,
+				WriteBufferSize: 512,
+			},
+			customRoundTripper: func(next http.RoundTripper) (http.RoundTripper, error) { return nil, errors.New("error") },
+			shouldError:        true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tt := componenttest.NewNopTelemetrySettings()
+			tt.TracerProvider = nil
```

Comment:
why?

Reply:
This is actually copied from `TestAllHTTPClientSettings()`. There are several `tt.TracerProvider = nil` lines in the unittests. I have dug through the Git history and think it's used to avoid the client's transport being wrapped in `otelhttp.Transport`. 🤔

Reply:
This is required by some testcases which need to check or modify the fields of `http.Transport`. This line is unnecessary here.